packages/java/endpoint/src/test/java/com/vaadin/hilla/EndpointControllerTest.java

```diff
@@ -445,6 +501,70 @@ public void should_NotCallMethod_When_DenyAll() {
                 .contains(EndpointAccessChecker.ACCESS_DENIED_MSG));
     }
 
+    @Test
+    public void should_AcceptMultipartFile() throws IOException {
+        // hilla request body
+        when(multipartRequest.getParameter(EndpointController.BODY_PART_NAME))
+                .thenReturn("{\"expectedLength\":5}");
+
+        // uploaded file
+        when(multipartRequest.getFileMap())
+                .thenReturn(Collections.singletonMap("/fileToCheck", multipartFile));
+
+        var vaadinController = createVaadinController(TEST_ENDPOINT);
+        var response = vaadinController.serveMultipartEndpoint(
+                TEST_ENDPOINT_NAME, "checkFileLength1", multipartRequest, null);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertTrue(response.getBody().contains("Check file length 1 OK"));
+
+        // check that the parameter order does not matter
+        response = vaadinController.serveMultipartEndpoint(TEST_ENDPOINT_NAME,
+                "checkFileLength2", multipartRequest, null);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertTrue(response.getBody().contains("Check file length 2 OK"));
+    }
+
+    @Test
+    public void should_AcceptMultipartFile_WithSingleParameter()
+            throws IOException {
+        // hilla request body
+        when(multipartRequest.getParameter(EndpointController.BODY_PART_NAME))
+                .thenReturn("{}");
+
+        // uploaded file
+        when(multipartRequest.getFileMap())
+                .thenReturn(Collections.singletonMap("/fileToCheck", multipartFile));
+
+        var vaadinController = createVaadinController(TEST_ENDPOINT);
+        var response = vaadinController.serveMultipartEndpoint(
+                TEST_ENDPOINT_NAME, "getFileLength", multipartRequest, null);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals("5", response.getBody());
+    }
+
+    @Test
+    public void should_AcceptMultipartFile_InComplexObjects()
+            throws IOException {
+        // hilla request body
+        when(multipartRequest.getParameter(EndpointController.BODY_PART_NAME))
+                .thenReturn(
+                        "{\"fileData\":{\"owner\":\"John\"},\"expectedLength\":5}");
+
+        // uploaded file
+        when(multipartRequest.getFileMap())
+                .thenReturn(Collections.singletonMap("/fileData/file", multipartFile));
+
+        var vaadinController = createVaadinController(TEST_ENDPOINT);
+        var response = vaadinController.serveMultipartEndpoint(
+                TEST_ENDPOINT_NAME, "checkOwnedFileLength", multipartRequest, null);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertTrue(response.getBody().contains("Check John's file length OK"));
+    }
+
```

Comment:
I suggest adding another test that verifies having multiple files as endpoint's params also works.